File: J_se_les12/HW/Core/Logic/Import/ImportBase.java
package Java.J_se_les12.HW.Core.Logic.Import;

import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public abstract class ImportBase {
    Phonebook phonebook;
    private String path;

    protected ImportBase(String path, Phonebook phonebook) {
        this.path = path;
        this.phonebook = phonebook;
    }

    public String getPath() {
        return this.path;
    }

    public abstract void importFiles();
}

File: J_se_les13/HW/Core/Logic/Import/ImportBase.cs
using HW.Core.Phonebook;
namespace HW.Core.Logic.Import
{
    public abstract class ImportBase
    {
        PhonebookV1 Phonebook { get; set; }
        public PhonebookV1 phonebook
        {
            get { return Phonebook; }
        }
        private string FilePath { get; set; }
        public string filePath
        {
            get { return FilePath; }
        }
        protected ImportBase(PhonebookV1 phonebook, string filePath)
        {
            Phonebook = phonebook;
            FilePath = filePath;

        }
        public abstract void ImportFiles();
    }
}